tasks/main.yml

```diff
@@ -1,42 +1,87 @@
 # SPDX-License-Identifier: MIT
 ---
-# Put the tasks for your role here.
-
-# An example of how to set distribution and version specific internal variables
-# (defined in vars/ directory)
-# see https://github.com/oasis-roles/meta_standards#platform-specific-variables
-- name: Set platform/version specific variables
-  include_vars: "{{ __template_vars_file }}"
-  loop:
-    - "{{ ansible_facts['os_family'] }}.yml"
-    - "{{ ansible_facts['distribution'] }}.yml"
-    - >-
-      {{ ansible_facts['distribution'] ~ '_' ~
-      ansible_facts['distribution_major_version'] }}.yml
-    - >-
-      {{ ansible_facts['distribution'] ~ '_' ~
-      ansible_facts['distribution_version'] }}.yml
-  vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
-
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: "Verify that the user accepts EULA variables"
+  assert:
+    that:
+      - mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula
+      - mssql_accept_microsoft_cli_utilities_for_sql_server_eula
+      - mssql_accept_microsoft_sql_server_2019_standard_eula
+    fail_msg:
+      - "You must accept EULA by setting the following variables to true:"
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+  register: __mssql_eula_evaluation
+
+- name: Configure the Microsoft SQL Server 2019 repository
+  yum_repository:
+    name: packages-microsoft-com-mssql-server-2019-gdr
+    description: Microsoft SQL Server 2019
+    baseurl: https://packages.microsoft.com/rhel/8/mssql-server-2019/
+    gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+    gpgcheck: true
+
+- name: Ensure that the Microsoft SQL Server package is installed
+  package:
+    name: "{{ __mssql_server_packages }}"
+    state: present
+
+- name: Configure the Microsoft SQL Server Tools repository
+  yum_repository:
+    name: packages-microsoft-com-prod
+    description: Microsoft SQL Server Tools
+    baseurl: https://packages.microsoft.com/rhel/8/prod/
+    gpgcheck: true
+
+- name: Ensure that SQL Server client tools are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __mssql_client_packages }}"
     state: present
+  environment:
+    - ACCEPT_EULA: Y
+
+- name: Gather system services facts
+  service_facts:
+
+- name: Set up MS SQL Server
+  command: /opt/mssql/bin/mssql-conf -n setup accept-eula
+  environment:
+    - ACCEPT_EULA: "Y"
+    - MSSQL_SA_PASSWORD: "{{ mssql_password }}"
+    - MSSQL_PID: "{{ mssql_edition }}"
+    - MSSQL_TCP_PORT: "{{ mssql_db_port }}"
+  when: |
+    "running" not in
+    ansible_facts["services"]["mssql-server.service"]["state"]
```

Comment:
I split this when statement into 3 lines to meet the requirement of max. 80 symbols per line. Is that fine though? This split kinda complicated the syntax

Reply:
This looks fine and it common to avoid the line length issue.  If you think this makes the syntax too complicated, please suppress the line length warning.  Most roles have at least 1 line or 1 file (or more) that we want to avoid line length checking.  https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.line_length and https://ansible-lint.readthedocs.io/en/latest/rules.html#false-positives-skipping-rules